### ISSUE TITLE
Rebuilding projections when there is zero events results in deadlock

### ISSUE THREAD

**Issue by invertedtomato** [JasperFx/marten#2519]
### Summary
Marten deadlocks when triggering a projection rebuild when there are no events in the database.

### Reproduction steps:
0. I'm using multi-tenancy with multiple databases on a single server (probably irrelevant)
1. Ensure you have at least one projection (and the projection is empty, though probably not relevant)
2. Ensure the `mt_events` table is empty
3. Trigger a projection rebuild on this projection. Something like:

```
     using var agent = await _documentStore.BuildProjectionDaemonAsync("core");
     await agent.RebuildProjection("projectprojection", TimeSpan.FromMinutes(1), CancellationToken.None);
```

### Expected result:
Nothing to happen, since there's no events there's nothing to rebuild with.

### Actual result:
Infinite loop in `HighWaterAgent.CheckNow()`

### Library version
Using Marten 5.11.0 from NuGet.

### Speculation on issue
It seems that `initialHighMark` is incorrectly resolving to `1` when there are no events. Should this be 0 instead?

**Comment by jeremydmiller** [comment]
@invertedtomato Sorry this one took so long to get to. Fix will be in 6.0.3 this week. Thanks for tracking that down.

**Comment by invertedtomato** [comment]
Thanks @jeremydmiller !
